src/main/java/net/minecraftforge/common/config/ConfigManager.java

```diff
@@ -161,10 +161,132 @@ public static void load(String modid, Config.Type type)
             }
         }
     }
+    
+    public static void updateConfig(String modid, Config.Type type)
+    {
+        FMLLog.fine("Attempting to update Configuration objects of mod %s for type %s", modid, type);
+        ClassLoader mcl = Loader.instance().getModClassLoader();
+        File configDir = Loader.instance().getConfigDir();
+        Multimap<Config.Type, ASMData> map = asm_data.get(modid);
+
+        if (map == null)
+            return;
+
+        for (ASMData targ : map.get(type))
+        {
+            try
+            {
+                Class<?> cls = Class.forName(targ.getClassName(), true, mcl);
+                String name = (String)targ.getAnnotationInfo().get("name");
+                if (name == null)
+                    name = modid;
+                String category = (String)targ.getAnnotationInfo().get("category");
+                if (category == null)
+                    category = "general";
+
+                File file = new File(configDir, name + ".cfg");
+
+                Configuration cfg = CONFIGS.get(file.getAbsolutePath());
+
+                updateConfig(cfg, cls, modid, type == Config.Type.INSTANCE, category);
+
+                cfg.save();
+
+            }
+            catch (Exception e)
+            {
+                FMLLog.log(Level.ERROR, e, "An error occurred trying to load a config for %s into %s", modid, targ.getClassName());
+                throw new LoaderException(e);
+            }
+        }
+    }
 
     // =======================================================
     //                    INTERNAL
     // =======================================================
+    static Configuration getConfiguration(String modid, String name) {
+        if(name == null || name.isEmpty())
+            name = modid;
+        File configDir = Loader.instance().getConfigDir();
+        File configFile = new File(configDir, name + ".cfg");
+        return CONFIGS.get(configFile.getAbsolutePath());
+    }
+    
+    private static void updateConfig(Configuration cfg, Class<?> cls, String modid, boolean isStatic, String category)
+    {
+        for (Field f : cls.getDeclaredFields())
+        {
+            if (!Modifier.isPublic(f.getModifiers()))
```

Comment:
First, I think I'm only exposing a method similar to what already is exposed (see code in sync()).
Second, isn't the point of all of this to get rid of getters and setters? The @Config annotation allows for mods to specify a class as config, which Forge then fills when sync is called. The point of this method (updateConfig()) is to notify forge, that the variables of the mod have been changed by other code and the internal Configuration object needs updating.